packages/ui/src/partials/react-form/form-field.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react"
+
+import {
+  Controller,
+  ControllerProps,
+  FieldPath,
+  FieldValues,
+  FormProvider,
+  useFormContext,
+} from "react-hook-form"
+import { FormItemContext } from "./form-item"
+
+export const Form = FormProvider
+
+/** 
+ * This is an extensible type that can be used to add additional properties to the form field context
+ * */ 
+type FormFieldContextValue<
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>
+> = {
+  name: TName
+}
+
+/**
+ * This is a context that is used to provide the form field context to the form field
+ *
+ * @default {} as FormFieldContextValue
+ */
+const FormFieldContext = React.createContext<FormFieldContextValue>(
```

Comment:
can the context be defined in one central react function and imported from there wherever required?

Reply:
This is not going to exported directly to be consumed. So is it worth it creating a separate file for this